agent.go

```diff
@@ -36,14 +36,23 @@ type Agent struct {
 
 // AgentEnv stores basic information of the endpoint
 type AgentEnv struct {
-	Init      string   `json:"init,omitempty"`
-	Ident     string   `json:"ident,omitempty"`
-	OS        string   `json:"os,omitempty"`
-	Arch      string   `json:"arch,omitempty"`
-	IsProxied bool     `json:"isproxied"`
-	Proxy     string   `json:"proxy,omitempty"`
-	Addresses []string `json:"addresses,omitempty"`
-	PublicIP  string   `json:"publicip,omitempty"`
+	Init      string      `json:"init,omitempty"`
+	Ident     string      `json:"ident,omitempty"`
+	OS        string      `json:"os,omitempty"`
+	Arch      string      `json:"arch,omitempty"`
+	IsProxied bool        `json:"isproxied"`
+	Proxy     string      `json:"proxy,omitempty"`
+	Addresses []string    `json:"addresses,omitempty"`
+	PublicIP  string      `json:"publicip,omitempty"`
+	AWS       AgentEnvAWS `json:"aws,omitempty"`
+}
+
+// Stores AWS specific agent environment values
+type AgentEnvAWS struct {
+	AWSInstanceID   string `json:"aws_instanceid,omitempty"`
+	AWSLocalIPV4    string `json:"aws_localipv4,omitempty"`
+	AWSAMIID        string `json:"aws_amiid,omitempty"`
+	AWSInstanceType string `json:"aws_instancetype,omitempty"`
```

Comment:
I'd drop the `aws_` prefixes in the variables and json, we're already under the `aws` key.